tools/generator/Java.Interop.Tools.Generator.ObjectModel/ClassGen.cs

```diff
@@ -244,113 +238,108 @@ public bool IsExplicitlyImplementedMethod (string sig)
 
 		public override void Generate (CodeGenerationOptions opt, GenerationInfo gen_info)
 		{
-			gen_info.CurrentType = FullName;
+			using (var sw = gen_info.OpenStream (opt.GetFileName (FullName))) {
+				sw.WriteLine ("using System;");
+				sw.WriteLine ("using System.Collections.Generic;");
+				sw.WriteLine ("using Android.Runtime;");
+				if (opt.CodeGenerationTarget != CodeGenerationTarget.XamarinAndroid) {
+					sw.WriteLine ("using Java.Interop;");
+				}
+				sw.WriteLine ();
+				sw.WriteLine ("namespace {0} {{", Namespace);
+				sw.WriteLine ();
 
-			StreamWriter sw = gen_info.Writer = gen_info.OpenStream(opt.GetFileName (FullName));
+				var generator = opt.CreateCodeGenerator (sw);
+				generator.WriteClass (this, "\t", gen_info);
 
-			sw.WriteLine ("using System;");
-			sw.WriteLine ("using System.Collections.Generic;");
-			sw.WriteLine ("using Android.Runtime;");
-			if (opt.CodeGenerationTarget != CodeGenerationTarget.XamarinAndroid) {
-				sw.WriteLine ("using Java.Interop;");
+				sw.WriteLine ("}");
 			}
-			sw.WriteLine ();
-			sw.WriteLine ("namespace {0} {{", Namespace);
-			sw.WriteLine ();
-
-			var generator = opt.CreateCodeGenerator (sw);
-			generator.WriteClass (this, "\t", gen_info);
-
-			sw.WriteLine ("}");
-			sw.Close ();
-			gen_info.Writer = null;
 		}
 
 		public static void GenerateTypeRegistrations (CodeGenerationOptions opt, GenerationInfo gen_info)
 		{
-			StreamWriter sw = gen_info.Writer = gen_info.OpenStream (opt.GetFileName ("Java.Interop.__TypeRegistrations"));
-
-			Dictionary<string, List<KeyValuePair<string, string>>> mapping = new Dictionary<string, List<KeyValuePair<string, string>>>();
-			foreach (KeyValuePair<string, string> reg in gen_info.TypeRegistrations) {
-				int ls          = reg.Key.LastIndexOf ('/');
-				string package  = ls >= 0 ? reg.Key.Substring (0, ls) : "";
-
-				if (JavaNativeTypeManager.ToCliType (reg.Key) == reg.Value)
-					continue;
-				List<KeyValuePair<string, string>> v;
-				if (!mapping.TryGetValue (package, out v))
-					mapping.Add (package, v = new List<KeyValuePair<string, string>>());
-				v.Add (new KeyValuePair<string, string>(reg.Key, reg.Value));
-			}
+			using (var sw = gen_info.OpenStream (opt.GetFileName ("Java.Interop.__TypeRegistrations"))) {
+
+				Dictionary<string, List<KeyValuePair<string, string>>> mapping = new Dictionary<string, List<KeyValuePair<string, string>>> ();
+				foreach (KeyValuePair<string, string> reg in gen_info.TypeRegistrations.OrderBy (p => p.Key, StringComparer.OrdinalIgnoreCase)) {
```

Comment:
Relatedly: the `.OrderBy()` is particularly useful now; see also: https://github.com/xamarin/java.interop/pull/442/files#r303698944